syncserver/src/tokenserver/handlers.rs

```diff
@@ -120,6 +121,24 @@ struct UserUpdates {
     uid: i64,
 }
 
+/// Optionally apply a timeout when running a future
+async fn apply_timeout<O, E>(
+    duration: Option<Duration>,
+    fut: impl std::future::Future<Output = Result<O, E>>,
+) -> Result<O, TokenserverError>
+where
+    E: Into<TokenserverError>,
+{
+    match duration {
+        Some(duration) => match timeout(duration, fut).await {
+            Ok(Ok(result)) => Ok(result),
+            Ok(Err(e)) => Err(e.into()),
+            Err(_) => Err(TokenserverError::elapsed()),
```

Comment:
Is this even needed w/ the `From` (maybe it still is since it's the outer exception)? If so we can probably just kill that `From` entirely.

Reply:
Ah, good catch. I was thinking we could keep it for safety reasons, but meh.